src/main/java/org/opensearch/jobscheduler/sweeper/JobSweeper.java

```diff
@@ -186,6 +197,82 @@ public void beforeClose() {
         this.fullSweepExecutor.shutdown();
     }
 
+    @Override
+    public Engine.Index preIndex(ShardId shardId, Engine.Index operation) {
+        if (JobSchedulerPlugin.GuiceHolder.getIndicesService() != null
```

Comment:
Can you invert the condition so that we exit early if any of these are not true? Makes it easer to read